2_Collections.py

```diff
@@ -0,0 +1,77 @@
+# Home task_2 Collections
+"""
+Write a code, which will:
+
+1. create a list of random number of dicts (from 2 to 10)
+dict's random numbers of keys should be letter,
+dict's values should be a number (0-100),
+example:[{'a': 5, 'b': 7, 'g': 11}, {'a': 3, 'c': 35, 'g': 42}]
+
+2. get previously generated list of dicts and create one common dict:
+if dicts have same key, we will take max value, and rename key with dict number with max value
+if key is only in one dict - take it as is,
+example:{'a_1': 5, 'b': 7, 'c': 35, 'g_2': 42}
+Each line of code should be commented with description.
+
+Commit script to git repository and provide link as home task result.
+"""
+
+# Task_1 Create list of 100 random numbers from 0 to 1000
+import random
+import string
+
+# Generate random number of list elements in the range from 2 to 10
+cnt_elements = random.randint(2, 10)
+
+dict_list = []
+
+# Create list of  dictionaries
+for i in range(cnt_elements):
+    # Generate number of elements in the list item. Each dictionary can contain from 2 to 10 elements
+    n = random.randint(2, 10)
+    # Initiate dictionary element
+    dict_elem = {}
+
+    # Create dictionary element
+    for j in range(n):
+        # Generate key and value for the dictionary element
+        k = random.choice(string.ascii_lowercase)
+        v = random.randint(0, 100)
+        # Add the new element with key k and value v
+        dict_elem[k] = v
+
+    dict_list.append(dict_elem)
+
+print(f"Number of list elements: {cnt_elements}")
+print(dict_list)
+
+# Task_2 Create one dictionary
+result_dict = {}
+
+# Counter to track the index of the dictionary
+index = 1
+
+# Iterate over the list of dictionaries
+for d in dict_list:
+    for key, value in d.items():
+        # Check if the key (or a variant with an underscore) already exists in result_dict
+        matching_keys = [k for k in result_dict if k.startswith(key)]
+
+        if matching_keys:
+            # Find the existing key that matches the current key
+            existing_key = matching_keys[0]
```

Comment:
The following logic must have some unexpected behaviour. Also, test your code with some simple tests. For Example:dict_list = [
    {'a': 1, 'b': 3, 'c': 1, 'e': 7},
    {'a': 2, 'b': 1, 'd': 9, 'f': 6},
    {'a': 4, 'b': 2, 'c': 1}]
It will return:
{'b': 3, 'c': 1, 'e': 7, 'd': 9, 'f': 6, 'a_3': 4}
Expected:
{'b_1': 3, 'c_1': 1, 'e': 7, 'd': 9, 'f': 6, 'a_3': 4}